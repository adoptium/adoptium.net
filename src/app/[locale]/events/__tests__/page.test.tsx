import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it } from "vitest"
import { axe } from "vitest-axe"
import Events from "../page"

describe("Events page", () => {
  it("renders correctly", () => {
    const { container } = render(<Events />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it("has no accessibility violations", async () => {
    const { container } = render(<Events />)
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })
})
