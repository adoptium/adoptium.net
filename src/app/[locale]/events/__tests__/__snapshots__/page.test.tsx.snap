// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Events page > renders correctly 1`] = `
<div>
  <div
    class="relative pt-20 pb-8 md:pb-16 md:pt-32 overflow-hidden"
  >
    <div
      class="w-full relative"
    >
      <div
        class="absolute z-[-1] w-[2396px] h-[1340px] left-[23px] md:left-[112px] top-[21px] md:top-[71px] bg-[#410170] shadow-[0_0_400px_rgba(65,1,112,1)] rounded-full blur-[400px]"
      />
      <div
        class="absolute z-[-1] w-[1487px] h-[893px] left-[120px] md:left-[676px] top-[60px] md:top-[195px] bg-[#B62987] shadow-[0_0_400px_rgba(182,41,135,1)] rounded-full blur-[400px]"
      />
      <div
        class="absolute z-[-1] w-[688px] h-[446px] left-[400px] md:left-[1131px] top-[318px] bg-[#FE8492] shadow-[0_0_400px_rgba(254,132,146,1)] rounded-full blur-[400px]"
      />
    </div>
    <div
      class="mx-auto max-w-[1048px] w-full px-6 lg:px-0 flex flex-col items-center justify-center"
    >
      <div
        class="self-stretch flex-col justify-center items-center gap-6 flex"
      >
        <div
          class="self-stretch flex-col justify-center items-center gap-4 flex"
        >
          <div
            class="justify-start items-center gap-3 inline-flex"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="#FF1464"
                height="8"
                rx="2"
                width="8"
                x="3"
                y="3"
              />
              <rect
                height="11"
                rx="3.5"
                stroke="#FF1464"
                stroke-opacity="0.25"
                stroke-width="3"
                width="11"
                x="1.5"
                y="1.5"
              />
            </svg>
            <div
              class="text-rose-600 text-base font-bold leading-normal"
            >
              Events
            </div>
          </div>
          <div
            class="self-stretch text-center text-white text-[56px] lg:text-[80px] leading-[114.286%] md:leading-[120%] font-semibold"
          >
            Eclipse  Foundation Events
          </div>
        </div>
        <div
          class="self-stretch text-center text-lightgrey text-[20px] font-normal leading-[140%] mx-auto max-w-[860px] px-2 w-full"
        >
          Follow the latest Eclipse Foundation Events
        </div>
      </div>
    </div>
  </div>
  <div
    class="bg-purple pt-8 px-4 max-w-[1000px] mx-auto"
  >
    <div
      class="mx-auto max-w-3xl px-6 lg:px-8 flex flex-col items-center justify-center mb-16"
    >
      <h2
        class="text-center text-4xl lg:text-5xl leading-[44px] lg:leading-[56px] font-semibold text-white-900"
      >
        Upcoming Events
      </h2>
    </div>
    <div
      class="relative"
    >
      <div
        class="flex justify-center items-center gap-8 md:gap-14 my-8"
      >
        <button
          class="flex items-center gap-3 px-3 py-2 rounded disabled:opacity-50"
          disabled=""
        >
          Previous
        </button>
        <div
          class="flex items-center gap-5"
        >
          <div
            class="flex items-center justify-between gap-6 border border-[#3E3355] rounded-[80px] px-6 py-3"
          />
        </div>
        <button
          class="flex items-center gap-3 px-3 py-2 rounded disabled:opacity-50"
        >
          Next
        </button>
      </div>
      <div
        class="flex items-center gap-5 mb-15 justify-center"
      >
        <a
          class="flex items-center gap-3"
          href="https://events.eclipse.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          View all Eclipse Foundation events
        </a>
        |
        <a
          class="flex items-center gap-3"
          href="https://newsroom.eclipse.org/node/add/events"
          rel="noopener noreferrer"
          target="_blank"
        >
          Submit an event
        </a>
      </div>
    </div>
  </div>
</div>
`;
